### ISSUE TITLE
Any chance for socket.io-p2p support?

### ISSUE THREAD

**Issue by mishoboss** [feathersjs/feathers-socketio#37]
I tried to make it work with https://github.com/socketio/socket.io-p2p, but socket.io-p2p creates another kind of object inside app.io object and feathers is unable to send/receive anything. Any chance for feathers-socketio to support socket.io-p2p too?

**Comment by daffl** [comment]
Can you post your setup that didn't work? I didn't try it but from the documentation it seems something like

```js
const socketio = require('feathers-socketio');
const p2p = require('socket.io-p2p-server').Server;

const app = feathers().configure(socketio(function(io) {
  io.use(p2p)
}));
```

On the server and on the client:

```js
const P2P = require('socket.io-p2p');
const io = require('socket.io-client');
const socket = io();
const p2p = new P2P(socket);

const app = feathers()
  .configure(feathers.socketio(p2p));
```

Should work.

**Comment by mishoboss** [comment]
Hi David, I just tried it exactly like you suggested. Client can't connect to server via websockets. Looks like the ws server isn't started anymore. And it is needed for the webrtc handshake.

**Comment by mishoboss** [comment]
Tried it exactly as you described and it works. Thank you! However server doesn't return responses. This is still websockets, haven't tried webrtc yet.

![image](https://cloud.githubusercontent.com/assets/1629929/18465047/4e25e020-799d-11e6-8113-6db0e0759cc1.png)

**Comment by mishoboss** [comment]
WebRTC works, but feathers services don't return responses again.

**Comment by daffl** [comment]
Can you share your app setup, maybe in a GitHub repository?

**Comment by ekryski** [comment]
@mishoboss yes please. I would be keen to see if we can get this working. 😁

**Comment by marshallswain** [comment]
Hey yeah. Nice. We could grow our family of supported providers.

**Comment by mishoboss** [comment]
Hi guys, my setup is pretty complex and still not fully working. The effort to make something reproducible by you is not small. But you don't need exactly my setup to reproduce it. Just try to use socket.io-p2p instead of socket.io on a clean feathers and you will see what I mean.

**Comment by domoticos** [comment]
Hello, have you reproduced it?

**Comment by mishoboss** [comment]
Hi, did you managed to reproduce it? Its really simple - just try to use socket.io-p2p instead of socket.io.

**Comment by frank-dspeed** [comment]
@mishoboss please prepare that simple repo for us i think we are doing enought work help is much wanted.

**Comment by daffl** [comment]
Although I really like the idea, this issue has been open for a while and I don't think we'll be able to look into it more any time soon.

**Comment by frank-dspeed** [comment]
@daffl i think you should reopen that as its easy to reproduce that demo and its a nice feature.

**Comment by mishoboss** [comment]
Hi guys. I also think this feature is not only a nice one, but also a very important one. It opens so much more use cases where Feathers could fit. 

BTW, I really can't understand why you need a simple example to reproduce it. Just set an empty feathers project and put socket.io-p2p in it the way it is described in its docs. Thats it. You will see the issue right after you do that. :)

**Comment by frank-dspeed** [comment]
@mishoboss the problem is why don't you do that :)

**Comment by marshallswain** [comment]
> Hi guys. I also think this feature is not only a nice one, but also a very important one. It opens so much more use cases where Feathers could fit.

I completely agree.

> BTW, I really can't understand why you need a simple example to reproduce it. Just set an empty feathers project and put socket.io-p2p in it the way it is described in its docs. Thats it. You will see the issue right after you do that. :)

It's mostly because of my schedule.  It's absolutely freaking packed like I never thought it could be.  I don't know what I expected with my life choices.  There are seven wild animals running around my house.  Five of them are my kids, and the other two are their parents.  We also run our own business, are employed full time and at the moment get to deal with the aftermath of what happens when houses flood.  ;)

Ok.  Joking aside, the little demo apps go a long way to save us time looking up things we've not yet experienced.  Each one of us has worked on Feathers on a 100% volunteer basis, and we bring our talents to the table and make cool stuff with it.  We're all also working on cool stuff in our spare time (which is unfortunately scarce) that will benefit the community in ways that each of us sees fit.  So far, our schedules don't interfere with doing amazing work when you look at the big picture.  We have each individually donated *thousands* of personal hours to the community.  And we definitely all try to foster a community where everyone's ideas are valued, encouraged, and where everybody feels like they have the support they need to work on their ideas.

In short, if you want to champion this feature, we've got your back.  But we have to take a supporting role at the moment, and not the lead.  If you want to take the lead at making this work, feel free to reopen and hit us up in chat with questions you have while you're figuring out how to make this feature work.  We've got your back, but you've gotta be in front.

By the way, we love you guys and the contributions you've already made in the community.  

❤️💯

Again, I think this is a sweet idea.
